### PR TITLE
Delete statements will not be created using `clear` 

### PR DESCRIPTION
**Pull request by ferrastas**
This is just a demo of an opened issue.

### CODE REVIEW

**Review by lcobucci — DISMISSED** [review]
Please provide a new test class with its own set of entities instead of modifying an existing one.

**Comment by ferrastas** [comment]
Yep, looks like a missing test case of the original fix :P

**Comment by ostrolucky** [comment]
I've pushed commit with better solution to fix #7758, without this regression

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH7761Test.php [hunk 1]
```diff
@@ -31,6 +31,15 @@ protected function setUp() : void
         $this->_em->clear();
     }
 
+    protected function tearDown()
+    {
+        $this->_em->clear();
```
Comment (lcobucci) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH7761Test.php]: This is already handled in the parent, so not needed :+1:

File: lib/Doctrine/ORM/UnitOfWork.php [hunk 2]
```diff
@@ -380,7 +380,12 @@ public function commit($entity = null)
         try {
             // Collection deletions (deletions of complete collections)
             foreach ($this->collectionDeletions as $collectionToDelete) {
-                $this->getCollectionPersister($collectionToDelete->getMapping())->delete($collectionToDelete);
+                if (!$collectionToDelete instanceof PersistentCollection ||
+                    // deferred explicit tracked collections can be removed only when owning relation was persisted
+                    $this->em->getClassMetadata(get_class($owner = $collectionToDelete->getOwner()))->isChangeTrackingDeferredImplicit() || 
+                    $this->isScheduledForDirtyCheck($owner)) {
+                    $this->getCollectionPersister($collectionToDelete->getMapping())->delete($collectionToDelete);
+                }
```
Comment (lcobucci) [inline lib/Doctrine/ORM/UnitOfWork.php]: I think there's a bit too much happening here. We can split things, do an early exit for non-persistent collections, and remove the assignment in condition.

**Review by lcobucci — CHANGES_REQUESTED** [review]
@ostrolucky thanks a lot for handling it. It's indeed a much better solution. I'll address these two tiny things and we're good to ship this.

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 2 above)
Comment (ostrolucky) [reply lib/Doctrine/ORM/UnitOfWork.php]: We can't do early exit, statement must be executed for non-persistent collections. And assignment needs to happen for persistent collections only.

Comment (lcobucci) [reply lib/Doctrine/ORM/UnitOfWork.php]: `continue` is an early-exit for the loop :)
I just pushed the changes :+1:

**Review by lcobucci — APPROVED** [review]

File: lib/Doctrine/ORM/UnitOfWork.php (hunk 2 above)
Comment (ostrolucky) [reply lib/Doctrine/ORM/UnitOfWork.php]: Yeah I know. I meant we can't do it without duplicating code. If that's fine, then ok

**Comment by lcobucci** [comment]
@ferrastas @ostrolucky thank you very much :+1:

**Comment by ferrastas** [comment]
Thanks to fix it @ostrolucky 🏆

**Pull request merged**